src/Algebra/Morphism/Structures.agda

```diff
@@ -68,6 +68,14 @@ module MagmaMorphisms (M₁ : RawMagma a ℓ₁) (M₂ : RawMagma b ℓ₂) wher
       ; surjective     = surjective
       }
 
+  IsSemigroupHomomorphism : (⟦_⟧ : A → B) → Set (a ⊔ ℓ₁ ⊔ ℓ₂)
+  IsSemigroupHomomorphism = IsMagmaHomomorphism
```

Comment:
I'm slightly reluctant to go down this path, as this opens up to us adding a crazy number of synonyms for *all* the structures/bundles in `Algebra.Bundles`. This is the same reason why we don't have `RawSemigroup`...

Reply:
So the funny reason for adding this one is that the *deprecation* warning for `IsSemingroupMorphism` mentioned `IsSemigroupHomomorphism`... even though it didn't in fact exist!

Thus I decided to make the deprecation warning less of a lie.

I'm fine with not adding a crazy number of synonyms.

Reply:
Ah my fault again. I guess we need to update the deprecation warning to `IsMagmaHomomorphism`...

Reply:
Done in PR #2344 . I'll also 'fix' this PR in that way, though I know I'll still need to split it.